src/spatialdata/_core/spatialdata.py

```diff
@@ -672,7 +671,7 @@ def filter_by_coordinate_system(
         # TODO: check whether full table dict should be returned or only those which annotate elements. Also check
         # filtering with tables having potentially different keys.
         if filter_tables:
-            tables = {}
+            tables: Tables = Tables(shared_keys=self._shared_keys)
```

Comment:
comparing this to #410 there is a difference here. However, I am not certain that `shared_keys` needs to be passed here since we are returning a new `SpatialData` object. What do you think @giovp @LucaMarconato ?

Reply:
yeah I agree, the shared keys should be instantiated in the return line 680. I think this can be closed!